src/test/Vector2Test.cpp

```diff
@@ -0,0 +1,384 @@
+	}
+};
+
+TEST_F(Vector2Test, Subtract) {
+	Vector2<int> result = int_vec1 - int_vec2;
+	EXPECT_EQ(result.x, -2);
+	EXPECT_EQ(result.y, -2);
+
+	Vector2<double> result2 = double_vec1 - double_vec2;
+	EXPECT_EQ(result2.x, -2.0);
+	EXPECT_EQ(result2.y, -2.0);
+
+	Vector2<long> result3 = long_vec1 - long_vec2;
+	EXPECT_EQ(result3.x, -2);
+	EXPECT_EQ(result3.y, -2);
+
+	Vector2<float> result4 = float_vec1 - float_vec2;
+	EXPECT_EQ(result4.x, -2.0f);
+	EXPECT_EQ(result4.y, -2.0f);
+}
+
+TEST_F(Vector2Test, SubtractScalar) {
+	Vector2<int> result = int_vec1 - 1;
+	EXPECT_EQ(result.x, 0);
+	EXPECT_EQ(result.y, 1);
+
+	Vector2<double> result2 = double_vec1 - 1.0;
+	EXPECT_EQ(result2.x, 0.0);
+	EXPECT_EQ(result2.y, 1.0);
+
+	Vector2<long> result3 = long_vec1 - 1;
+	EXPECT_EQ(result3.x, 0);
+	EXPECT_EQ(result3.y, 1);
+
+	Vector2<float> result4 = float_vec1 - 1.0f;
+	EXPECT_EQ(result4.x, 0.0f);
+	EXPECT_EQ(result4.y, 1.0f);
+}
+
+TEST_F(Vector2Test, Add) {
+	Vector2<int> result = int_vec1 + int_vec2;
+	EXPECT_EQ(result.x, 4);
+	EXPECT_EQ(result.y, 6);
+
+	Vector2<double> result2 = double_vec1 + double_vec2;
+	EXPECT_EQ(result2.x, 4.0);
+	EXPECT_EQ(result2.y, 6.0);
+
+	Vector2<long> result3 = long_vec1 + long_vec2;
+	EXPECT_EQ(result3.x, 4);
+	EXPECT_EQ(result3.y, 6);
+
+	Vector2<float> result4 = float_vec1 + float_vec2;
+	EXPECT_EQ(result4.x, 4.0f);
+	EXPECT_EQ(result4.y, 6.0f);
+}
+
+TEST_F(Vector2Test, AddScalar) {
+	Vector2<int> result = int_vec1 + 1;
+	EXPECT_EQ(result.x, 2);
+	EXPECT_EQ(result.y, 3);
+
+	Vector2<double> result2 = double_vec1 + 1.0;
+	EXPECT_EQ(result2.x, 2.0);
+	EXPECT_EQ(result2.y, 3.0);
+
+	Vector2<long> result3 = long_vec1 + 1;
+	EXPECT_EQ(result3.x, 2);
+	EXPECT_EQ(result3.y, 3);
+
+	Vector2<float> result4 = float_vec1 + 1.0f;
+	EXPECT_EQ(result4.x, 2.0f);
+	EXPECT_EQ(result4.y, 3.0f);
+}
+
+TEST_F(Vector2Test, Multiply) {
+	Vector2<int> result = int_vec1 * int_vec2;
+	EXPECT_EQ(result.x, 3);
+	EXPECT_EQ(result.y, 8);
+
+	Vector2<double> result2 = double_vec1 * double_vec2;
+	EXPECT_EQ(result2.x, 3.0);
+	EXPECT_EQ(result2.y, 8.0);
+
+	Vector2<long> result3 = long_vec1 * long_vec2;
+	EXPECT_EQ(result3.x, 3);
+	EXPECT_EQ(result3.y, 8);
+
+	Vector2<float> result4 = float_vec1 * float_vec2;
+	EXPECT_EQ(result4.x, 3.0f);
+	EXPECT_EQ(result4.y, 8.0f);
+}
+
+TEST_F(Vector2Test, MultiplyScalar) {
+	Vector2<int> result = int_vec1 * 2;
+	EXPECT_EQ(result.x, 2);
+	EXPECT_EQ(result.y, 4);
+
+	Vector2<double> result2 = double_vec1 * 2.0;
+	EXPECT_EQ(result2.x, 2.0);
+	EXPECT_EQ(result2.y, 4.0);
+
+	Vector2<long> result3 = long_vec1 * 2;
+	EXPECT_EQ(result3.x, 2);
+	EXPECT_EQ(result3.y, 4);
+
+	Vector2<float> result4 = float_vec1 * 2.0f;
+	EXPECT_EQ(result4.x, 2.0f);
+	EXPECT_EQ(result4.y, 4.0f);
+}
+
+TEST_F(Vector2Test, Divide) {
+	Vector2<int> result = int_vec1 / int_vec2;
+	EXPECT_EQ(result.x, 0);
+	EXPECT_EQ(result.y, 0);
+
+	Vector2<double> result2 = double_vec1 / double_vec2;
+	EXPECT_EQ(result2.x, 0.33333333333333331);
```

Comment:
Voor float comparison hoor je eigenlijk [`EXPECT_FLOAT_EQ`](https://google.github.io/googletest/reference/assertions.html#EXPECT_FLOAT_EQ) of [`EXPECT_NEAR`](https://google.github.io/googletest/reference/assertions.html#EXPECT_NEAR) te gebruiken (ivm afrondingsfouten)